plugins/module_utils/azure_rm_common_ext.py

```diff
@@ -213,44 +257,54 @@ def default_compare(self, modifiers, new, old, path, result):
+        if new_managed_type == 'None' and curr_managed_type == 'None':
             pass
+
+        changed = False
         # If type set to None, and Resource has current identities, remove UserAssigned identities
-        elif param_identity.get('type', 'None') == 'None':
-            changed = True
+        # Or
         # If type in module args different from current type, update identities
-        elif current_managed_type != param_identity.get('type', 'None'):
+        if new_managed_type == 'None' or curr_managed_type != new_managed_type:
             changed = True
 
+        curr_user_assigned_identities = set(curr_identity.get('user_assigned_identities', {}).keys())
+        new_user_assigned_identities = set(new_identity.get('user_assigned_identities', {}).get('id', []))
+        result_user_assigned_identities = new_user_assigned_identities
+
         # If type in module args contains 'UserAssigned'
-        if 'UserAssigned' in param_identity.get('type', 'None'):
-            if param_identity.get('user_assigned_identities', {}).get('append', False) is True:
-                external_identities = param_identities.union(current_managed_identities)
-                if len(current_managed_identities) != len(external_identities):
-                    # update identities
-                    changed = True
-            # If new identities have to overwrite current identities
-            else:
-                # Check if module args identities are different as current ones
-                if current_managed_identities.difference(external_identities) != set():
-                    changed = True
+        if allow_identities_append and \
+           'UserAssigned' in new_managed_type and \
+           new_identity.get('user_assigned_identities', {}).get('append', True) is True:
+            result_user_assigned_identities = new_user_assigned_identities.union(curr_user_assigned_identities)
+
+        # Check if module args identities are different as current ones
+        if curr_user_assigned_identities.difference(result_user_assigned_identities) != set():
+            changed = True
 
-        new_identity = self.managed_identity['identity'](type=param_identity.get('type'))
+        result_identity = self.managed_identity['identity'](type=new_identity.get('type'))
 
         # Append identities to the model
-        if external_identities:
-            new_identity.user_assigned_identities = {}
-            for identity in external_identities:
-                new_identity.user_assigned_identities[identity] = self.managed_identity['user_assigned']()
+        if len(result_user_assigned_identities) > 0:
+            result_identity.user_assigned_identities = {}
+            for identity in result_user_assigned_identities:
+                result_identity.user_assigned_identities[identity] = self.managed_identity['user_assigned']()
 
-        return changed, new_identity
+        return changed, result_identity
```

Comment:
For backwards compat
```
+
+    update_identities = update_managed_identity
```

Reply:
I added this with a comment to remove it after all modules are changed to the new name